contentcuration/contentcuration/frontend/shared/utils.js

```diff
@@ -178,3 +180,25 @@ export function isSuccessor({ rootVertex, vertexToCheck, graph, immediateOnly =
   const blackVertices = coloredVertices.filter(v => v.color === BLACK);
   return blackVertices.map(v => v.name).includes(vertexToCheck);
 }
+
+/**
+ * Chunks an array, calling `callback` with `chunkSize` amount of items,
+ * expecting callback to return `Promise` that when resolved will allow
+ * next chunk to be processed.
+ *
+ * @param {mixed[]} things
+ * @param {number} chunkSize
+ * @param {Function} callback
+ * @return {Promise<mixed[]>}
+ */
+export function promiseChunk(things, chunkSize, callback) {
```

Comment:
Since this chunks the array, I still think `Chunk` is appropriate. Why do you ask?

Reply:
Ah ok! I was a little confused as to what it meant like "a chunk of an individual promise"? I wonder if it would help to name it something like `chunkPromises` or `promiseChunks`, but not a blocker

Reply:
Oh gotcha. Yeah I guess the naming of this could be confusing any which way. I think I just need more comments, especially doc comments.